source/Octopus.Cli/Commands/Package/PackCommand.cs

```diff
@@ -31,36 +31,36 @@ public class PackCommand : CommandBase, ICommand, ISupportFormattedOutput
+            zip.Add<PackageCompressionLevel>("compressionLevel=", "[Optional] Set compression level of package: none, fast, optimal (default).", c => packageCompressionLevel = c);
 
             var nuget = Options.For("NuGet packages");
-            nuget.Add("author=", "[Optional, Multiple] Add an author to the package metadata; defaults to the current user", v => authors.Add(v));
-            nuget.Add("title=", "[Optional] The title of the package", v => title = v);
-            nuget.Add("description=", "[Optional] A description of the package; defaults to a generic description", v => description = v);
-            nuget.Add("releaseNotes=", "[Optional] Release notes for this version of the package", v => releaseNotes = v);
-            nuget.Add("releaseNotesFile=", "[Optional] A file containing release notes for this version of the package", v => releaseNotesFile = v);
+            nuget.Add<string>("author=", "[Optional, Multiple] Add an author to the package metadata; defaults to the current user", v => authors.Add(v));
+            nuget.Add<string>("title=", "[Optional] The title of the package", v => title = v);
+            nuget.Add<string>("description=", "[Optional] A description of the package; defaults to a generic description", v => description = v);
+            nuget.Add<string>("releaseNotes=", "[Optional] Release notes for this version of the package", v => releaseNotes = v);
+            nuget.Add<string>("releaseNotesFile=", "[Optional] A file containing release notes for this version of the package", v => releaseNotesFile = v);
             
             var basic = Options.For("Basic options");
-            basic.Add("id=", "The ID of the package; e.g. MyCompany.MyApp", v => id = v);
-            basic.Add("format=", "Package format. Options are: NuPkg, Zip. Defaults to NuPkg, though we recommend Zip going forward", fmt => packageBuilder = SelectFormat(fmt));
-            basic.Add("version=", "[Optional] The version of the package; must be a valid SemVer; defaults to a timestamp-based version", v => version = string.IsNullOrWhiteSpace(v) ? null : new SemanticVersion(v));
-            basic.Add("outFolder=", "[Optional] The folder into which the generated NuPkg file will be written; defaults to '.'", v => { v.CheckForIllegalPathCharacters(nameof(outFolder)); outFolder = v;});
-            basic.Add("basePath=", "[Optional] The root folder containing files and folders to pack; defaults to '.'", v => { v.CheckForIllegalPathCharacters(nameof(basePath)); basePath = v;});
-            basic.Add("verbose", "[Optional] verbose output", v => verbose = true);
+            basic.Add<string>("id=", "The ID of the package; e.g. MyCompany.MyApp", v => id = v);
+            basic.Add<PackageFormat>("format=", "Package format. Options are: NuPkg, Zip. Defaults to NuPkg, though we recommend Zip going forward", fmt => packageBuilder = SelectFormat(fmt));
```

Comment:
Should the list of valid values be generated from the enum now? E.g. https://github.com/OctopusDeploy/OctopusCLI/blob/7cbb503ac8720481735bf62b5b004719a5722178/source/Octopus.Cli/Commands/InstallAutoCompleteCommand.cs#L25